hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieWriteConfig.java

```diff
@@ -2855,8 +2863,12 @@ public Integer getWritesFileIdEncoding() {
     return props.getInteger(WRITES_FILEID_ENCODING, HoodieMetadataPayload.RECORD_INDEX_FIELD_FILEID_ENCODING_UUID);
   }
 
-  public boolean needResolveWriteConflict(WriteOperationType operationType) {
+  public boolean needResolveWriteConflict(WriteOperationType operationType, boolean isMetadataTable, boolean streamingWritesToMetadataTableEnabled) {
     WriteConcurrencyMode mode = getWriteConcurrencyMode();
+    if (isMetadataTable && streamingWritesToMetadataTableEnabled) {
+      // for metadata table, if streaming writes are enabled, we might need to perform conflict resolution.
+      return true;
```

Comment:
Why does the conflict resolution differ from the data table NBCC? We should add some notes here.